oss-internship-2020/pffft/main_pffft_sandboxed.cc

```diff
@@ -0,0 +1,203 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <glog/logging.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/times.h>
+#include <syscall.h>
+#include <time.h>
+
+#include <cassert>
+#include <cmath>
+#include <cstdio>
+
+#include "pffft_sapi.sapi.h"
+#include "sandboxed_api/util/flag.h"
+#include "sandboxed_api/vars.h"
+
+ABSL_DECLARE_FLAG(string, sandbox2_danger_danger_permit_all);
+ABSL_DECLARE_FLAG(string, sandbox2_danger_danger_permit_all_and_log);
+
+class PffftSapiSandbox : public PffftSandbox {
+ public:
+  std::unique_ptr<sandbox2::Policy> ModifyPolicy(
+      sandbox2::PolicyBuilder*) {
+    return sandbox2::PolicyBuilder()
+        .AllowStaticStartup()
+        .AllowOpen()
+        .AllowRead()
+        .AllowWrite()
+        .AllowSystemMalloc()
+        .AllowExit()
+        .AllowSyscalls({
+            __NR_futex,
+            __NR_close,
+            __NR_getrusage,
+        })
+        .BuildOrDie();
+  }
+};
+
+double UclockSec() { return static_cast<double>(clock()) / CLOCKS_PER_SEC; }
+
+int array_output_format = 0;
```

Comment:
AFAICT, this is never modified anywhere. Perhaps we should turn it into a command-line flag?